### PR TITLE
Fix incorrect types in #state{} record

### PR DESCRIPTION
**Pull request by jesperes**

### CODE REVIEW

File: src/supervisor3.erl [hunk 1]
```diff
@@ -172,11 +172,11 @@
 -endif.
 
 -record(state, {name,
-                strategy               :: strategy(),
```
Comment (k32) [inline src/supervisor3.erl:175]: Please move all field specs from the record definition to the definition of `-type state() :: ...`. This code change makes it legal that "half of record can be initialized", but as far as I follow the code, it shouldn't be legal. If I follow the code correctly, either all fields should be `undefined` (when `init` callback returns `post_init`), or all fields have valid values.

**Review by k32 — CHANGES_REQUESTED** [review]

**Comment by k32** [comment]
P.S. since you're here, can you perhaps get rid of `-ifdef(use_specs)` stuff (i.e. always use specs)? It will eliminate annoying "no behavior information found for supervisor3" warnings.

**Comment by jesperes** [comment]
> P.S. since you're here, can you perhaps get rid of `-ifdef(use_specs)` code? It will eliminate annoying "no behavior information found for supervisor3" warnings.

Our fork is pretty far behind upstreams, so not sure how much we should be diverging.

**Comment by k32** [comment]
Upstream is _completely_ unrelated to our fork (it's the whole rabbitmq) :) We shouldn't treat this repo as a fork.

**Comment by mikpe** [comment]
Please remove the internal ticket reference from the commit message.

**Pull request closed without merging**